Softeq.XToolkit.Common/Commands/AsyncCommand.cs

```diff
@@ -2,158 +2,109 @@
 // http://www.softeq.com
 
 using System;
-using System.Reflection;
 using System.Threading.Tasks;
-using System.Windows.Input;
 using Softeq.XToolkit.Common.Extensions;
-using Softeq.XToolkit.Common.Weak;
 
 namespace Softeq.XToolkit.Common.Commands
 {
     /// <summary>
-    ///     A command whose sole purpose is to relay its functionality to other
-    ///     objects by invoking async delegates. The default return value for the CanExecute
-    ///     method is 'true'.  This class does not allow you to accept command parameters in the
-    ///     Execute and CanExecute callback methods.
+    ///     An implementation of <see cref="IAsyncCommand"/>.
+    ///     Allows Commands to safely be used asynchronously with Task.
     /// </summary>
-    public abstract class AsyncCommandBase : ICommand
+    public class AsyncCommand : AsyncCommandBase, IAsyncCommand
     {
-        private readonly WeakFunc<bool>? _canExecute;
-        private bool _isRunning;
-
-        protected AsyncCommandBase(Func<bool>? canExecute)
-        {
-            if (canExecute != null)
-            {
-                _canExecute = new WeakFunc<bool>(canExecute);
-            }
-        }
+        private readonly Func<object, Task> _execute;
```

Comment:
I think, not in this time, because it will be broke behaviors the async commands for all the places where delegate may be released.

Good point for the next PRs.